pkg/xds/server/v3/reconcile.go

```diff
@@ -192,8 +192,8 @@ func (s *templateSnapshotGenerator) GenerateSnapshot(ctx xds_context.Context, pr
 }
 
 type snapshotCacher interface {
```

Comment:
Is it possible to get rid of this interface and `simpleSnapshotCacher` type? I think we could use `envoy_cache.SnapshotCache` directly where it's needed.

Reply:
Probably, but I'd prefer to work on it in another PR if that's ok.

Reply:
The upstream interfaces are pretty wonky and inconsistent. For example we need `simpleSnapshotCacher` because of https://github.com/envoyproxy/go-control-plane/issues/302

Reply:
Right, it just seems like sometimes we build a node object just to pass it to the interface –https://github.com/kumahq/kuma/blob/master/pkg/xds/server/v3/reconcile.go#L56. But I agree it should be out of the scope of this PR

Reply:
https://github.com/kumahq/kuma/issues/4926